Campus Density/DiningCells/DetailControllerHeaderCell.swift

```diff
@@ -64,58 +63,37 @@ class DetailControllerHeaderCell: UICollectionViewCell {
         fatalError("init(coder:) has not been implemented")
     }
 
-    func configure(with place: Place) {
-        self.place = place
-        displayLabel.text = place.displayName
-        if place.displayName.count >= 23 {
+    func configure(displayName: String, hours: [DailyInfo]) {
+        displayLabel.text = displayName
+        if displayName.count >= 23 {
             displayLabel.font = .twentyEightBold
         }
 
-        if place.isClosed {
-            timeString = NSMutableAttributedString(string: "Closed")
-            timeString.addAttribute(.foregroundColor, value: UIColor.orangeyRed, range: NSRange(location: 0, length: timeString.mutableString.length))
-        } else {
-            let closingString = getNextClosingText()
-            timeString = NSMutableAttributedString(string: "Open until \(closingString)")
-            timeString.addAttribute(.foregroundColor, value: UIColor.lightTeal, range: NSRange(location: 0, length: timeString.mutableString.length))
-        }
-
-        timeLabel.attributedText = timeString
+        let timeText = getNextClosingText(hours: hours)
+        let attributes = [NSAttributedString.Key.foregroundColor: (timeText == "Closed" ? UIColor.orangeyRed : UIColor.lightTeal)]
+        timeLabel.attributedText = NSMutableAttributedString(string: timeText, attributes: attributes)
     }
 
     /**
-     Returns a string representing the next closing time for this place, with the hh:mm aa format.
+     Returns a user-facing string representing the next closing time for this place.
```

Comment:
Great PR! Maybe just clarify what this means, but really like the work you've done.